bot/modals/class_modal.py

```diff
@@ -0,0 +1,164 @@
+from typing import Union
+
+import discord
+from discord import Interaction, TextStyle
+from discord.ui import Modal, TextInput
+
+from bot.consts import Colors
+from bot.data.class_repository import ClassRepository
+from bot.messaging.events import Events
+from bot.models.class_models import ClassChannelScaffold
+from bot.sock_bot import SockBot
+from bot.utils.helpers import error_embed
+
+# Min and max of major length, i.e., 'CPSC', 'HCC', 'MATH'
+MIN_MAJOR_LEN = 3
+MAX_MAJOR_LEN = 4
+# Min and max of class title length, i.e., 'Data Structures and Algorithms', etc.
+MIN_TITLE_LEN = 5
+MAX_TITLE_LEN = 50
+# Min and max of class instructor, i.e., 'Dean', 'Widman', 'VanScoy', etc.
+MIN_INSTR_LEN = 2
+MAX_INSTR_LEN = 15
+# Max length of class description, i.e., 'Students learn about...'
+MAX_DESCR_LEN = 500
+# Min and max of class number, i.e., 1060, 4910, 8400, etc.
+MIN_CLASS_NUM = 1000
+MAX_CLASS_NUM = 8999
+
+
+class AddClassModal(Modal):
+
+    major = TextInput(
+        label='Course Major',
+        default='cpsc',
+        placeholder='cpsc',
+        min_length=MIN_MAJOR_LEN,
+        max_length=MAX_MAJOR_LEN,
+        row=0
+    )
+    course_number = TextInput(
+        label='Course Number',
+        placeholder='2120',
+        min_length=4,
+        max_length=4,
+        row=1
+    )
+    course_title = TextInput(
+        label='Course Name',
+        placeholder='Data Structures & Algorithms',
+        min_length=MIN_TITLE_LEN,
+        max_length=MAX_TITLE_LEN,
+        row=2
+    )
+    professor = TextInput(
+        label='Course Instructor',
+        placeholder='Dean',
+        min_length=MIN_INSTR_LEN,
+        max_length=MAX_INSTR_LEN,
+        row=3
+    )
+    course_description = TextInput(
+        label='Course Description',
+        placeholder='Students learn about...',
+        style=TextStyle.long,
+        max_length=MAX_DESCR_LEN,
+        required=False,
+        row=4
+    )
+
+    def __init__(self, bot: SockBot,
+                 *,
+                 class_data: tuple[str, int] | None = None,
+                 channel: discord.TextChannel | None = None,
+                 role: discord.Role | None = None):
+        super().__init__(title='📔 Add Class' if not channel else f'Insert #{channel.name}')
+        self._bot = bot
```

Comment:
Is a `bot` attribute not already set by the superclass?

Reply:
Nah Modals do not have a `bot` attribute, so in order to call the events I needed to pass the bot instance to it.